### DISCUSSION TITLE
ÖPNVKarte link alternative

### DISCUSSION THREAD

**Discussion by mpeter50** [streetcomplete/StreetComplete#3971] [Feedback]
Hi!

Today I've found a potential problem, and wanted to ask you what do you think of it.

The user gets [a link](https://öpnvkarte.de) to ÖPNVKarte's website in one of the achievements.
When they open the link, in some cases strange things can happen with it.

One case is that because of the `ö` character, the web browser might display `http://xn--pnvkarte-m4a.de` instead, in some cases even in the address bar, but on error pages always (like the 'unable to connect' page of Firefox):

![image](https://user-images.githubusercontent.com/83356418/163857566-66356442-9351-4a18-9482-8038066467c2.png)

An other case is that software like Pi-hole sometimes seem not to be able to resolve addresses with accent characters:

![image](https://user-images.githubusercontent.com/83356418/163858140-39ddfe7a-6bb6-456a-b80d-6798ed852ff9.png)

But when the browser finally makes the query with the domain name in punycode form, then the user's DNS filter might step in the way: sometimes domains starting with `xn--` are blocked by the user('s network admin) in the local DNS resolver because unicode characters are mostly used in phishing links.

![image](https://user-images.githubusercontent.com/83356418/163860031-158c488d-9698-44da-8ea3-47cefdb4aec4.png)

Sure, the user can whitelist this one domain, but only if they realize that the website might be fine, and it's actually the local DNS server that blocked the connection.

I've found https://openbusmap.org/ ([source](https://wiki.openstreetmap.org/wiki/%C3%96PNVKarte#%C3%96pnvkarte_elsewhere)), which seems to be the same service, but if you replace the link please check it to be sure.
An other reason for replacing the link might also be that this other link is easier to remember, and type on most keyboards.

What do you think?

**Comment by HolgerJeromin** [comment]
If someone is configuring his network [Pi-hole] to drop some 12 year old standard this is his own fault IMO and his reponsibility.

**Comment by matkoniecz** [comment]
> Pi-hole sometimes seem not to be able to resolve addresses with accent characters

This is a Pi-hole issue, not StreetComplete issue

> user's DNS filter might step in the way

It is a DNS filter issue, not StreetComplete issue.

> I've found https://openbusmap.org/ ([source](https://wiki.openstreetmap.org/wiki/%C3%96PNVKarte#%C3%96pnvkarte_elsewhere)), which seems to be the same service

Is it official ÖPNVKarte service or endorsed by them?

**Reply by mpeter50** [reply]
I can't find any sources that would confirm this is official. Also, the user who added it to the mentioned wiki page did not explain where did he find it.
Additionally, it looks this domain name points to a different IP, and has bigger latency, so this is probably not an official domain.

Well, I stopped thinking this change is a good idea.

**Reply by mnalis** [reply]
@mpeter50 you should probably contact the author of that blacklist, and suggest to them to add that specific legitimate domain to their whitelist.
